### PR TITLE
Fix #76: Enhance Clickable Area and Hover Responsiveness in Donations Table

### LINKED ISSUE #76: Enhance Clickable Area and Hover Responsiveness in Donations Table
To improve the usability of the /donations page, we propose making the entire table cell clickable, not just the donation ID, and adding responsiveness on hover.

image

Responsibilities:

- Clickable Table Cells: Extend the clickable area from just the donation ID to the entire table cell corresponding to each donation. This change should apply to cells that lead to specific donor detail pages.
- Hover Responsiveness: Implement a hover effect for the clickable cells to indicate to users that they are interactive. This could include a slight color change or underlining text to visually signify the cell's functionality.

### PR DESCRIPTION
**Pull request by Gayatrinj**
**Fixes #76 **

### What was changed?

- Enhanced the donations table to make entire rows clickable instead of just the item ID.
- Added hover effects to improve interactivity and user experience.
- Prevented undesired row navigation when interacting with certain elements (e.g., checkboxes and buttons).
- Added CSS to implement hover effects.
- Implemented test cases to validate the hover functionality and clickable row behavior.

### Why was it changed?

- Previously, only the item ID was clickable, leading to a poor user experience.
- Making rows clickable provides a more intuitive and user-friendly interface.
- Hover effects visually indicate that rows are interactive, improving usability.
- Test cases ensure reliability and avoid regressions in future updates.

### How was it changed?

- **Script Changes**:
  Added an onClick event listener to <tr> elements to navigate to the donation details page when rows are clicked:
```
onClick={() => navigate(`/donations/${item.id}`)}
```

- **Used stopPropagation for buttons and checkboxes to prevent triggering the row click event:
```
e.stopPropagation();

```
**CSS changes**:

- Added hover effects for rows using the _.clickable-row_ class

### Screenshots that show the changes:

- **Before:**

- Only the item ID was clickable.
- No hover effect to indicate interactivity.
-
before

- **After:**

Entire row is clickable.
Added hover effect to rows for better feedback.

now

### CODE REVIEW

**Comment by Anjali0407-git** [comment]
Changes looks good. 

Please address the following changes before merging this PR:

1. There were some latest changes in the upstream branch, can you please pull them and fix merge conflicts if any?
2. It seems you did not run the prettier check on your code. Please follow this documentation to ensure prettier checks are passed https://github.com/oss-slu/material-donor-mutual-assist/blob/main/README.md#code-formatting

File: client-app/src/Components/DonatedItemsList.tsx [hunk 1]
```diff
@@ -362,70 +362,63 @@ const DonatedItemsList: React.FC = () => {
...
-                                        }
-                                    />
-                                </td>
-                            )}
-                        </tr>
-                    ))}
-                </tbody>
-            </table>
+    <thead>
+        <tr>
```
Comment (Anjali0407-git) [inline client-app/src/Components/DonatedItemsList.tsx:366]: It seems these are the trivial changes with some indentation removed. Prettier should apply the formatting updates to your code. Please run prettier write on your code

**Review by Anjali0407-git — CHANGES_REQUESTED** [review]

**Comment by Anjali0407-git** [comment]
You can see in github actions that Prettier check pipeline failed 
https://github.com/oss-slu/material-donor-mutual-assist/actions

**Comment by Gayatrinj** [comment]
I have completed all the requested changes:

- Merged the latest changes from the main branch and resolved any conflicts.
- Verified and formatted the code using Prettier (prettier:write and prettier:check).

File: client-app/package.json [hunk 2]
```diff
@@ -1,95 +1,94 @@
 {
-  "name": "my-app",
```
Comment (Anjali0407-git) [inline client-app/package.json:2]: @Gayatrinj `package.json` and `package-lock.json` file should not auto formatted by prettier as you can see [here ](https://github.com/oss-slu/material-donor-mutual-assist/blob/main/.prettierignore)that these files are ignored by prettier. Please do not push any trivial changes to `package.json` and `package-lock.json` file unless you have some new dependency added to it.

File: client-app/.babelrc [hunk 3]
```diff
@@ -1,6 +1,3 @@
 {
-    "presets": [
-        "@babel/preset-env",
-        "@babel/preset-react"
-    ]
-}
```
Comment (Anjali0407-git) [inline client-app/.babelrc:6]: @Gayatrinj Please revert the changes to this file. I would suggest to not push any trivial changes.

File: client-app/package.json (hunk 2 above)
Comment (Anjali0407-git) [reply client-app/package.json:2]: Please revert changes to both of these files and push only if you have any new dependency added

**Review by Anjali0407-git — CHANGES_REQUESTED** [review]

File: client-app/src/Components/Register.js [hunk 4]
```diff
@@ -32,146 +33,117 @@ const Register = () => {
             setErrorMessage('Password must be at least 5 characters');
             return;
         }
-
```
Comment (Anjali0407-git) [inline client-app/src/Components/Register.js:35]: Can you also check why others changes are appearing in your PR? As you have already merged the upstream main branch into your working branch, these changes of others are not supposed to be shown here.

**Review by Anjali0407-git — CHANGES_REQUESTED** [review]

**Pull request closed without merging**